docs/onboarding-guide.md

```diff
@@ -0,0 +1,171 @@
+
+
+### [AWS.Deploy.Common](https://github.com/aws/aws-dotnet-deploy/tree/main/src/AWS.Deploy.Common)
+
+This component contains POCO objects and common logic that is used by other components in the deployment tool
+**Note** - This project does not reference any other project except for `../src/AWS.Deploy.Constants/AWS.Deploy.Constants.projitems`
+
+#### Key Components
+
+* [**ProjectDefinitionParser**](https://github.com/aws/aws-dotnet-deploy/blob/main/src/AWS.Deploy.Common/ProjectDefinitionParser.cs) - This class contains the logic to validate whether a .NET project exists at the path specified via `dotnet aws deploy —project-path <PROJECT-PATH>`. It also parses the user’s csproj/fsproj file and returns a [ProjectDefinition](https://github.com/aws/aws-dotnet-deploy/blob/main/src/AWS.Deploy.Common/ProjectDefinition.cs)
+* **[RecipeDefinition](https://github.com/aws/aws-dotnet-deploy/tree/main/src/AWS.Deploy.Recipes/RecipeDefinitions)** - This is a POCO class that serves as the deserialized model for all recipe files found [here](https://github.com/aws/aws-dotnet-deploy/tree/main/src/AWS.Deploy.Recipes/RecipeDefinitions)
+* **[Recommendation](https://github.com/aws/aws-dotnet-deploy/blob/main/src/AWS.Deploy.Common/Recommendation.cs)** - This is a POCO class that stores different metadata such as the [RecipeDefinition](https://github.com/aws/aws-dotnet-deploy/tree/main/src/AWS.Deploy.Recipes/RecipeDefinitions) and [ProjectDefinition](https://github.com/aws/aws-dotnet-deploy/blob/main/src/AWS.Deploy.Common/ProjectDefinition.cs). There is 1:1 mapping between each recipe file and a recommendation object. Each recommendation object has a computed priority which determines its precedence.
+* [**DeploymentBundleDefinition**](https://github.com/aws/aws-dotnet-deploy/blob/main/src/AWS.Deploy.Common/DeploymentBundles/DeploymentBundleDefinition.cs) - This is a POCO class that serves as the deserialized model for all [DeploymentBundleDefinitions](https://github.com/aws/aws-dotnet-deploy/tree/main/src/AWS.Deploy.Recipes/DeploymentBundleDefinitions)
+* [**OptionSettingItem**](https://github.com/aws/aws-dotnet-deploy/blob/main/src/AWS.Deploy.Common/Recipes/OptionSettingItem.cs) - This is a POCO class that serves as the deserialized model for option settings found inside [deployment recipes](https://github.com/aws/aws-dotnet-deploy/blob/66fb1e34bbea10bbd995596e1eec801b79ff9102/src/AWS.Deploy.Recipes/RecipeDefinitions/aws-deploy-recipe-schema.json#L467). These settings can be tweaked by the user to customize their deployment. This is defined as a a partial class and the rest of its functionality is specified inside [OptionSettingItem.ValueOverride](https://github.com/aws/aws-dotnet-deploy/blob/66fb1e34bbea10bbd995596e1eec801b79ff9102/src/AWS.Deploy.Common/Recipes/OptionSettingItem.ValueOverride.cs)
```

Comment:
There are a couple links to specific commits that need to be cleaned up in this whole section.